lib/api/core/hotelClerk.js

```diff
@@ -578,13 +604,13 @@ function cleanUpRooms (roomId) {
     this.kuzzle.dsl.removeRoom(this.rooms[roomId])
       .then(() => {
         this.kuzzle.pluginsManager.trigger('room:remove', roomId);
+        delete this.rooms[roomId];
         deferred.resolve(roomId);
       })
       .catch(error => {
         this.kuzzle.pluginsManager.trigger('log:error', error);
         deferred.reject(error);
-      })
-      .finally(() => delete this.rooms[roomId]);
+      });
```

Comment:
While merging it seems you rolled back a bugfix: the room deletion should occur in the ``finally`` statement to make sure it is deleted even if ``dsl.removeRoom`` fails.